### ISSUE TITLE
Basic Data Export

### ISSUE THREAD

**Issue by jonnalley** [HHS/OPRE-OPS#3067]
### User Story
As an OPS System Admin, I want to access and download a data export of all attributes of relevant OPS entities so that I can perform obligatory actions on the data outside of OPS. 

### Acceptance Criteria
- [x] User can procure and download a set of all relevant entries in for a given set of concepts such as budget lines, contracts, projects, etc....
- [ ] ~This export should only be accessible to user with the System Owner role.~

-----------------------------------------

#### UX Tasks
- [x] Identify and prioritize all the places/OPS entities & data types that need to be exported (List of CANs, budget lines, agreements, and ?)
- [x] Work with OPRE to identify which columns are required
- [x] Design a way to trigger the export process
- [x] Share designs ready for dev
- [x] Update the rest of figma workflows to match new design (agreements list, BLs, CANs list)

##### FE
- [x] Export agreements list
- [x] Export budget lines

### Definition of Done Checklist
- [x] UI works as designed (UX team)
- [x] PR(s) have been merged to main
- [x] Design/tech debt eliminated
- [x] New design/tech debt documented (if applicable)
- [x] Build process updated
- [x] Documentation updated or added
- [x] Feature flags/toggles created

### Additional Context & Resources
* This might be necessary due to certain OPS features not being released yet.  In example, getting data on contracts now since IAAs and AAs aren't ready
* This would also be necessary to aggregate info from disparate system that OPS could not potentially do 
* Another reason might be to supplement additional information outside of OPS for common upstream data gathering.

### Links
- consider using [react-csv](https://github.com/react-csv/react-csv): A widely-used library for generating and downloading CSV files in React. It provides two main components: CSVLink (for clickable downloads) and CSVDownload (for automatic downloads).

**Comment by kimschulke** [comment]
@weimiao67 @Santi-3rd updated designs are ready for dev on export -- let us know what you think. We decided to use an export link and changed the filter button to a link because 2 buttons side by side looked a little cluttered.

https://www.figma.com/design/dsQFe6G2KjcTWNq7NNZujm/OPRE-OPS?node-id=17408-36809&m=dev

**Comment by jonnalley** [comment]
@kimschulke we had documented on Tuesday that they said they do not need a separate export of cans, at least for right now. 🤷

**Comment by kimschulke** [comment]
updated the designs in figma for Agreements list, BLs list and CANs list...

we can remove export from CANs list if not needed

**Comment by kimschulke** [comment]
@jonnalley here are my notes from tuesday

Sheila notes there will be 2 different exports
- $/money export
- Procurement export


**BLs list export / money export**
- BL Amount
- BL Fee
- Procurement shop w/ fee rate
- Dont need the BL total

Need to add BL columns for:

- Research type: applied vs evaluative
- New or continuing
- In OPS, this is defined at the agreement level and applied to all BLs on that agreement
IAAs dont have new vs continuing…

**Agreements list / procurement export**
Knowing we dont have the award data in OPS yet, start with:
- Agreement name
- Agreement total
- Contract type (firm fixed price, etc)
- Who the vendor is
- Who the COR is

Long-term procurement goals:
- Agreement metadata and bottom line of the agreement
- What’s been obligated to date and etc
- Something for contracts, something for grants?

Sample request on how this spreadsheet might be used —
I need all the contracts from Mathematica

**Comment by jonnalley** [comment]
@kimschulke For the BL export it's currently doing:

"BL ID #", "Agreement", "SC", "Obligate By", "FY", "CAN", "SubTotal", "Fees", "Status"]

so based on the comment above,  it looks like we're missing the "procurement shop with fee rate",    which I presume is meant to be two separate columns??  @weimiao67 @Santi-3rd 


@kimschulke For the agreement export, it's currently doing:

"Agreement", "Project", "Type", "Agreement SubTotal",  "Agreement Fees",  "Next Budget Line SubTotal",   "Next Budget Line Fees",  "Next Obligate By"

So looks like we're missing vendor and COR at least @weimiao67 @Santi-3rd

**Comment by weimiao67** [comment]
Yes, I will add `procurement shop with fee rate` to BL export, and add vendor and COR to agreement table

**Comment by kimschulke** [comment]
Demos looks great! Just need to add the Research Type and New vs Continuing columns once we have that ready to go

**Comment by jonnalley** [comment]
@kimschulke based on the previous comment, I will presume you're comfortable with the first DoD item being checked. I do believe we have some documentation to update so I'm working that angle now.

**Comment by jonnalley** [comment]
@kimschulke @andressacurvelo @npage123  I saw the google doc/PDF of the how-to guides and FAQ are updated to reflect that exporting is now possible, thank you! I thought I remembered that we were holding off on updating the in-app version of those in the Help Center but are you all comfortable enough with the current state to check off the DoD item for documentation being updated for this story ?

**Comment by npage123** [comment]
I am comfortable with checking off the documentation DoD unless I'm missing something @kimschulke @andressacurvelo?
